### DISCUSSION TITLE
Reverse Slider (CTRL + G) Screwed up the slider body and slider algorithm in Edit mode

### DISCUSSION THREAD

**Discussion by timelessnesses** [ppy/osu#16334] [Q&A] [answered]
Here's the footage

https://user-images.githubusercontent.com/59832159/148224697-fa4b3351-cc39-4583-8eea-93b0e8f49d08.mp4

**Comment by integerrr** [comment]
Sounds like #15084 - has this been regressed somehow?

**Reply by timelessnesses** [reply]
idk but latest osu!lazer did

**Comment by peppy** [comment]
Please attach the beatmap and provide the time value. Video is not required.

**Reply by bdach** [reply]
beatmap is https://osu.ppy.sh/beatmapsets/362286#osu/3402199 - looking into it

**Comment by bdach** [comment]
so this is happening because the slider in question is snapped to 1/8, while the map itself has a snap divisor of 1/4 set globally:

![2022-01-06-134125_1250x147_scrot](https://user-images.githubusercontent.com/20418176/148384624-cbb31977-2240-4b19-933e-b9754ca2b590.png)
![2022-01-06-134134_1238x145_scrot](https://user-images.githubusercontent.com/20418176/148384629-a268e913-e41a-44eb-8ccd-04ef1e215c33.png)

this makes it so that the following logic kicks in on reverse:

https://github.com/ppy/osu/blob/a9e4a0fd61459233bd30617ad5a775707ed51105/osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderSelectionBlueprint.cs#L247

snapping the slider back an eighth. not too sure what the expected behaviour there is.

**Reply by timelessnesses** [reply]
the expect behaviour should be same kinda osu! stable did in edit mode

**Reply by bdach** [reply]
well the question i'd like to know the answer to is why objects are snapped to 1/8 on a 1/4 map. is this common?

**Reply by peppy** [reply]
Very common. Saying a map is a "1/4 map" doesn't really make sense (the snap divisor is generally something you change constantly during the mapping process). Weird that that would change the slider length when reversing thought, should not be the case.

**Reply by bdach** [reply]
well the thing is that this is a 1% case where the snapping isn't desirable. as is, it is applied on a high enough level wherein it applies to *all* operations on the slider path. and in the 99% of cases, that's fine (e.g. it's fine that moving a control point snaps to the current beat snap divisor) - but this is the one case where this isn't needed. i guess this means the snapping should maybe be somewhere lower down but this in turn creates a worry that some operation will make it possible to unsnap a slider entirely.

**Reply by peppy** [reply]
I can't reproduce this even when changing the snap divisor. Is there something else to it?

**Reply by bdach** [reply]
i don't... think so? reproduces instantly for me for any slider that isn't snapped to the current beat divisor (e.g. `00:11:407 (1)`)

**Comment by bdach** [comment, marked as answer]
#16354 would address this (if it passes review).
